src/main/scala/nl.knaw.dans.dccd.convert/DccdConvertExportApp.scala

```diff
@@ -15,10 +15,216 @@
+  var language: String = _
+  var isoDefault: Locale = _
+  var languageISO3: String = _
+
+  var ddmAccessRights: String = _
+  var now: Date = _
+  var ddmAvailable: String = _
+  var ddmAudience: String = _
+
+
+  def createCsvReportFromXmlFiles(dir: String): String = {
+    dirPath = getPath(dir)
+    dirList = getListOfSubDirectories(getPath(dir).toString).toList
+    dirList.foreach { i =>
+      dirr = dirPath.toString + "/" + i + "/administrative/project_metadata.xml"
+      dirr2 = dirPath.toString + "/" + i + "/administrative/user.xml"
+      println(dirr)
+      metadata = XML.loadFile(dirr.toString)
+      parsedMetadata = parseNoWS(metadata.toString())
+      user = XML.loadFile(dirr2.toString)
+      parsedUser = parseNoWS(user.toString())
+
+      dataset = dirr
+      dcTitle = (parsedMetadata \\ "title").text
+      depositorId = (parsedUser \\ "id").text
+      dcxCreatorOrganization = (parsedMetadata \\ "ownerOrganizationId").text
+
+      date = (parsedMetadata \\ "stateChanged").text
+      parsedDate = Format.parseObject(date)
+      ddmCreated = Format.format(parsedDate)
+
+      dctRightsHolder = dcxCreatorOrganization
+      dcDescription = "Dendrochronological project"
+      dcSubject = "Dendrochronology"
+      dcElementType = "dccd_element_type: " + (parsedMetadata \\ "elementType").text
+      dcObjectType = "dccd_object_type: " + (parsedMetadata \\ "objectType").text
+      dcTypeMetadata = "dccd_type: " + (parsedMetadata \\ "type").text
+      dcCategory = "dccd_category: " + (parsedMetadata \\ "category").text
+      dcTaxon = "dccd_taxon: " + (parsedMetadata \\ "taxon").text
+
+      def timeRange(parsedData: NodeSeq): String = {
+        val parsedData = parsedMetadata
+        val first = (parsedData \\ "firstYear").text
+        val second = (parsedData \\ "lastYear").text
+        if (first.contains("-").equals(false) && second.contains("-").equals(false)) {
+          range = first + "-" + second + " AD"
+        }
+        if (first.contains("-") && second.contains("-")) {
+          range = ((first.toInt - 1) * (-1)).toString + "-" + ((second.toInt - 1) * (-1)).toString + " BC"
+        }
+        if (first.contains("-") && second.contains("-").equals(false)) {
+          range = ((first.toInt - 1) * (-1)).toString + " BC" + "-" + second + " AD"
+        }
+        range
+      }
+
+      dctTemporal = timeRange(parsedMetadata)
+
+      dcxSpatialX = (parsedMetadata \\ "lat").text
+      dcxSpatialY = (parsedMetadata \\ "lng").text
+
+      dcIdentifierSid = (parsedMetadata \\ "sid").text + "\n"
+      dcIdentifierIdentifier = (parsedMetadata \\ "identifier").text
+
+      dcxRelationQualifier = " 'references' "
+      dcxRelationTitle = "Digital Collaboratory for Cultural Dendrochronology (DCCD)"
+      dcxRelationLink = "https://dendro.dans.knaw.nl/dccd/project/" + (parsedMetadata \\ "sid").text
+
+      dcxRelationQualifier_2 = " 'IsFormatOf' "
+      dcxRelationTitle_2 = (parsedMetadata \\ "sid").text
```

Comment:
You use sid a lot, extract it into a val and use that one. Also we could fail if we don't have a sid?